var/spack/repos/builtin/packages/go-bootstrap/package.py

```diff
@@ -18,50 +19,54 @@ class GoBootstrap(Package):
-        url="https://dl.google.com/go/go1.4-bootstrap-20171003.tar.gz",
-    )
-    version(
-        "1.4-bootstrap-20170531",
-        sha256="49f806f66762077861b7de7081f586995940772d29d4c45068c134441a743fa2",
-        url="https://storage.googleapis.com/golang/go1.4-bootstrap-20170531.tar.gz",
-    )
-    version(
-        "1.4-bootstrap-20161024",
-        sha256="398c70d9d10541ba9352974cc585c43220b6d8dbcd804ba2c9bd2fbf35fab286",
-        url="https://storage.googleapis.com/golang/go1.4-bootstrap-20161024.tar.gz",
-    )
-
-    provides("golang@:1.4-bootstrap-20171003")
+    maintainers("alecbcs")
 
     depends_on("git", type=("build", "link", "run"))
 
-    conflicts(
-        "os=monterey",
-        msg="go-bootstrap won't build on MacOS Monterey: "
-        "try `brew install go` and `spack external find go`",
-    )
-    conflicts("target=aarch64:", msg="Go bootstrap doesn't support aarch64 architectures")
-
-    # This virtual package allows a fallback to gccgo for aarch64,
-    # where go-bootstrap cannot be built(aarch64 was added with Go 1.5)
-    provides("go-external-or-gccgo-bootstrap")
-
-    # Support for aarch64 was added in Go 1.5, use an external package or gccgo instead:
-    conflicts("@:1.4", when="target=aarch64:")
-
     executables = ["^go$"]
 
+    # List binary go releases for multiple operating systems and architectures.
+    # These binary versions are not intended to stay up-to-date. Instead we
+    # should update these binary releases on a yearly schedule as
+    # bootstrapping requirements are modified by new releases of go.
+    go_releases = {
+        "1.17.13": {
+            "darwin": {
+                "amd64": "c101beaa232e0f448fab692dc036cd6b4677091ff89c4889cc8754b1b29c6608",
+                "arm64": "e4ccc9c082d91eaa0b866078b591fc97d24b91495f12deb3dd2d8eda4e55a6ea",
+            },
+            "linux": {
+                "amd64": "4cdd2bc664724dc7db94ad51b503512c5ae7220951cac568120f64f8e94399fc",
+                "arm64": "914daad3f011cc2014dea799bb7490442677e4ad6de0b2ac3ded6cee7e3f493d",
+                "ppc64le": "bd0763fb130f8412672ffe1e4a8e65888ebe2419e5caa9a67ac21e8c298aa254",
+            },
+        }
+    }
+
+    # Normalize architectures returned by platform to those used by the
+    # Go project.
+    go_targets = {
+        "aarch64": "arm64",
+        "arm64": "arm64",
+        "ppc64le": "ppc64le",
+        "amd64": "amd64",
+        "x86_64": "amd64",
+    }
+
+    # determine system os and architecture/target
+    os = platform.system().lower()
+    target = go_targets[platform.machine().lower()]
+
+    # construct releases for current system configuration
+    for release in go_releases:
+        if os in go_releases[release] and target in go_releases[release][os]:
+            version(release, sha256=go_releases[release][os][target])
+            provides("go-or-gccgo-bootstrap@{}".format(release), when="@{}".format(release))
```

Comment:
```suggestion
            provides(f"go-or-gccgo-bootstrap@{release}", when=f"@{release}")
```
just slightly easier to read

Reply:
Wait we can use f-strings now?!? I'd been holding off on them since I wasn't sure if the minimum python version Spack supported was new enough to use f-strings.